### ISSUE TITLE
I retrained on the cifar dataset and got a low map value

### ISSUE THREAD

**Issue by wjpapple** [htconquer/ddh#3]
Hi,I made my own training set(50000) and test set(10000) on the cifar dataset.The features are extracted from resnet50,I retrained on your code,but I got the low map value.The 16bits is 0.2328.The 32bits is 0.2694.
And do you know why?

**Comment by ht014** [comment]
Do you create the similarity matrix?

**Comment by wjpapple** [comment]
I create the similarity matrix,but I don't know how you construct the similarity matrix.Do you preprocess the  feature vector?

**Comment by ht014** [comment]
You can run construct_s.py to create a similarity matrix but you need to modify the input file path.

**Comment by wjpapple** [comment]
Sorry,I don't know how do you build the cifar_KNN.npz.I get the feature vector from resnet50,then I create the cifar_kNN.npz,but I get a different result.

**Comment by wjpapple** [comment]
I create the cifar_KNN.npz.The code is shown below.
from sklearn.metrics.pairwise import cosine_similarity
import scipy.io as scio
import numpy as np
def normData(dataset):
    maxVals = dataset.max(axis=0)
    minVals = dataset.min(axis=0)
    ranges = maxVals - minVals
    retData = (dataset - minVals) / ranges
    return retData
data = scio.loadmat("cifar_train_resnet.mat")#The features are extracted from resnet50
# print(dict(data))
train_feature=data['train_feature']
#s=np.zeros((50000,50000))
feature=normData(train_feature)
s=cosine_similarity(feature)
print(s.shape)
np.save("cifar_similarity.npy",s)
a=np.zeros((50000,1000))
for i in range(50000):
    b=np.argsort(-s[i])
    a[i]=b[0:1000]
    print(i)
a=a.astype(np.int)
np.save("cifar_KNN.npy",a)#cifar_knn
print(a)

**Comment by ht014** [comment]
cifar_KNN.npz stores each point's KNNs ID.

**Comment by wjpapple** [comment]
yes,I know.Do you use k1=15, k2 =6 for experiment?It seems that k1,k2 is set to a higher value to achieve the effect.

**Comment by ht014** [comment]
Yes, actually it depends on the feature you extracted. When the constructed similarity matrix is more accurate, the performance is higher.

**Comment by wjpapple** [comment]
How many training sets(cifar) are there in your paper?

**Comment by ht014** [comment]
Actually,  it's the size of 'S', 10000.
